### ISSUE TITLE
Pass CancellationToken To RunAsync and ReadAsync

### ISSUE THREAD

**Issue by khalidabuhakmeh** [adamralph/simple-exec#230]
## Use case(s)

Some processes are long-running tasks, and the ability to `kill` these processes might be a good thing. I have a process that starts playing an audio file and I'd like to stop the audio with a cancellationToken.

## Description

```
            var audio = await SimpleExec.Command.ReadAsync(
                "afplay",
                string.Join(" ", arguments),
                noEcho: true,
                cancellationToken: cancellationToken
            );
```

The issue is that the cancellation request needs to call `process.Kill` or else the process continues to run in the background.

## Alternatives

Handroll starting a process.

## Additional context

I'll try and make an attempt myself at implementing the change, but my `async` foo is mediocre at best.

**Comment by adamralph** [comment]
Re-opening this, to consider whether `Run` and `Read` should also accept a `CancellationToken` as per @danielmarbach's suggestion in https://github.com/adamralph/simple-exec/pull/238.

**Comment by adamralph** [comment]
[Decided not to support `CancellationToken` in the sync methods.](https://github.com/adamralph/simple-exec/pull/242#issuecomment-739317834)

**Comment by adamralph** [comment]
Released in [6.4.0-rc.1](https://www.nuget.org/packages/SimpleExec/6.4.0-rc.1).

@khalidabuhakmeh please give it a try.
